com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/adapter/handler/ConfigurationDoneRequestHandler.java

```diff
@@ -82,7 +82,7 @@ private void handleDebugEvent(DebugEvent debugEvent, IDebugSession debugSession,
         } else if (event instanceof VMDeathEvent) {
             vmHandler.disconnectVirtualMachine(event.virtualMachine());
             context.setVmTerminated();
-            context.getProtocolServer().sendEvent(new Events.ExitedEvent(0));
+            context.getProtocolServer().sendEvent(new Events.ExitedEvent(event.virtualMachine().process().exitValue()));
```

Comment:
`event.virtualMachine().process()` might return `null`, so a null check is necessary. Additionally, I'm investigating how the exit mechanism works, but it seems my debug session never receives a VMDeathEvent when the process exits. Instead, only a VMDisconnectEvent occurs.